packages/@lwc/engine-dom/src/apis/hydrate-component.ts

```diff
@@ -33,42 +66,43 @@ export function hydrateComponent(
         );
     }
 
+    if (getAssociatedVMIfPresent(element)) {
+        /* eslint-disable-next-line no-console */
+        console.warn(
```

Comment:
Why limit this to a warning and not throw an error?

Reply:
stated in [the RFC](https://github.com/salesforce/lwc-rfcs/blob/master/text/0117-ssr-rehydration.md#customelementsdefine) (second bullet).